CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+# Contributing guide
+
+Welcome to the project, and thanks for considering contributing to this project. 
+This document should answer many questions you may have about contributing to this project.
+
+If you have any questions or need clarifications on topics covered here, please feel free to reach out to us on the [#cnoe-interest](https://cloud-native.slack.com/archives/C05TN9WFN5S) channel on CNCF Slack.
+
+## Setting up a development environment
+
+To get started with the project on your machine, you need to install the following tools:
+1. Go 1.21+. See [this official guide](https://go.dev/doc/install) from Go authors.
+2. Make. You can install it through a package manager on your system. E.g. Install `build-essential` for Ubuntu systems.
+3. Docker. Similar to Make, you can install it through your package manager.
+
+Once required tools are installed, clone this repository. `git clone https://github.com/cnoe-io/idpbuilder.git`.
+
+Then change your current working directory to the repository root. e.g. `cd idpbuilder`.
+
+All subsequent commands described in this document assumes they are executed from the repository root.
+
+## Building from the main branch
+
+1. Checkout the main branch. `git checkout main`
+2. Build the binary. `make build`. This compiles the project. It will take several minutes for the first time. Example output shown below:
+    ```
+    ~/idpbuilder$ make build
+    test -s /home/ubuntu/idpbuilder/bin/controller-gen && /home/ubuntu/idpbuilder/bin/controller-gen --version | grep -q v0.12.0 || \
+    GOBIN=/home/ubuntu/idpbuilder/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.12.0
+    /home/ubuntu/idpbuilder/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./api/..." output:crd:artifacts:config=pkg/controllers/resources
+    /home/ubuntu/idpbuilder/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
+    go fmt ./...
+    go vet ./...
+    go build -o idpbuilder main.go  
+    ```
```

Comment:
I think the indentation for triple backticks is off, no? should start from the beginning of the line

Reply:
If it starts from the beginning, the code block doesn't line up well with bullet points.